schematic/store/synapse.py

```diff
@@ -1343,15 +1364,32 @@ def add_entities(
             manifest (pd.DataFrame): modified to add entitiyId as appropriate.
 
         '''
+
+        # Expected behavior is to annotate files if `Filename` is present regardless of `-mrt` setting
+        if 'filename' in [col.lower() for col in manifest.columns]:
+            # get current list of files and store as dataframe
+            dataset_files = self.getFilesInStorageDataset(datasetId)
+            files = self._get_file_entityIDs(manifest=manifest,dataset_files=dataset_files, only_new_files=False)
+            file_df = pd.DataFrame(files)
+            
+            # Merge dataframes to add entityIds
+            manifest = manifest.merge(file_df, how = 'left', on='Filename', suffixes=['_x',None]).drop('entityId_x',axis=1)
+
+
         for idx, row in manifest.iterrows():
-            if not row["entityId"] and (manifest_record_type == 'file_and_entities' or 
-                manifest_record_type == 'table_file_and_entities'):
-                manifest, entityId = self._create_entity_id(idx, row, manifest, datasetId)
-            elif not row["entityId"] and manifest_record_type == 'table_and_file':
-                # If not using entityIds, fill with manifest_table_id so 
-                row["entityId"] = manifest_synapse_table_id
-                manifest.loc[idx, "entityId"] = manifest_synapse_table_id
-                entityId = ''
+            if 'filename' not in [col.lower() for col in manifest.columns]:
```

Comment:
@GiaJordan I tried commented out this if statement and the else statement, and it seems like the annotation could still be added correctly. The code that you added from 1368-1376 is sufficient to fix the issue

Reply:
The bug was caused by assigning the `manifest_synapse_table_id` to the `entityId` column for every row. Adding in this line (in addition to line 1369: `if 'filename' in [col.lower() for col in manifest.columns]:`) completes the behavior split for file based and non-file based metadata. We still want the last `else` clause to trigger so that the annotations get assigned appropriately for each row/file.